src/anomalib/models/image/dinomaly/lightning_model.py

```diff
@@ -0,0 +1,471 @@
+            raise
+        else:
+            return batch.update(pred_score=predictions.pred_score, anomaly_map=predictions.anomaly_map)
+
+    def configure_optimizers(self) -> OptimizerLRScheduler:
+        """Configure optimizer and learning rate scheduler for Dinomaly training.
+
+        Sets up the training configuration with frozen DINOv2 encoder and trainable
+        bottleneck and decoder components. Uses StableAdamW optimizer with warm
+        cosine learning rate scheduling.
+
+        The total number of training steps is determined dynamically from the trainer
+        configuration, supporting both max_steps and max_epochs settings.
+
+        Returns:
+            OptimizerLRScheduler: Tuple containing optimizer and scheduler configurations.
+
+        Raises:
+            ValueError: If neither max_epochs nor max_steps is defined.
+
+        Note:
+            - DINOv2 encoder parameters are frozen to preserve pre-trained features
+            - Only bottleneck MLP and decoder parameters are trained
+            - Uses truncated normal initialization for Linear layers
+            - Learning rate schedule: warmup (100 steps) + cosine decay
+            - Base learning rate: 2e-3, final learning rate: 2e-4
+            - Total steps determined from trainer's max_steps or max_epochs
+        """
+        # Freeze all parameters
+        for param in self.model.parameters():
+            param.requires_grad = False
+        # Unfreeze bottleneck and decoder
+        for param in self.model.bottleneck.parameters():
+            param.requires_grad = True
+        for param in self.model.decoder.parameters():
+            param.requires_grad = True
+
+        trainable = torch.nn.ModuleList([self.model.bottleneck, self.model.decoder])
+        self._initialize_trainable_modules(trainable)
+
+        # Determine total training steps dynamically from trainer configuration
+        # Check if trainer has valid max_epochs and max_steps set
+        max_epochs = getattr(self.trainer, "max_epochs", -1)
+        max_steps = getattr(self.trainer, "max_steps", -1)
+
+        if max_epochs is None:
+            max_epochs = -1
+        if max_steps is None:
+            max_steps = -1
+
+        if max_epochs < 0 and max_steps < 0:
+            msg = "A finite number of steps or epochs must be defined"
+            raise ValueError(msg)
+
+        if max_epochs < 0:
+            # max_epochs not set, use max_steps directly
+            total_steps = max_steps
+        elif max_steps < 0:
+            # max_steps not set, calculate from max_epochs
+            total_steps = max_epochs * len(self.trainer.datamodule.train_dataloader())
+        else:
+            # Both are set, use the minimum (training stops at whichever comes first)
+            total_steps = min(
+                max_steps,
+                max_epochs * len(self.trainer.datamodule.train_dataloader()),
+            )
+
+        optimizer_config = TRAINING_CONFIG["optimizer"]
```

Comment:
@ashwinvaidya17, can you confirm if this dict-based config is API/CLI friendly ?